File: Flix.Backend/Flix.Services/Database/MovieIssueReport.cs
using Flix.Model.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Flix.Services.Database
{
    public class MovieIssueReport
    {
        [Key]
        public int Id { get; set; }

        public int MovieId { get; set; }
        [ForeignKey(nameof(MovieId))]
        public Movie Movie { get; set; } = null!;

        public int ReportedByUserId { get; set; }
        [ForeignKey(nameof(ReportedByUserId))]
        public User ReportedBy { get; set; } = null!;
        [Required]
        [MaxLength(150)]
        public string Header {get; set;} = string.Empty;
        [MaxLength(2000)]
        public string? Description { get; set; } = string.Empty;
        public ReportStatus Status { get; set; } = ReportStatus.Open;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public int? ReviewedByUserId { get; set; }
        [ForeignKey(nameof(ReviewedByUserId))]
        public User? ReviewedBy { get; set; }
        public DateTime? ResolvedAt { get; set; }
        [MaxLength(2000)]
        public string? AdminComment { get; set; }
    }
}

File: Flix.UI/flix_mobile/lib/models/movie_issue_report.dart
import 'package:flix_mobile/enums/report_status.dart';
import 'package:flix_mobile/models/movie.dart';
import 'package:flix_mobile/models/user.dart';
import 'package:json_annotation/json_annotation.dart';

part 'movie_issue_report.g.dart';

@JsonSerializable()
class MovieIssueReport {
  MovieIssueReport(
    this.id,
    this.movieId,
    this.header,
    this.description,
    this.status,
    this.createdAt,
    this.reportedBy,
    this.movie,
    this.reviewedBy,
    this.resolvedAt,
    this.adminComment,
  );

  final int? id;
  final int? movieId;

  final String? header;
  final String? description;

  final ReportStatus? status;
  final DateTime? createdAt;

  final User? reportedBy;
  final Movie? movie;
  final User? reviewedBy;

  final DateTime? resolvedAt;
  final String? adminComment;

  String? get statusLabel => status == null ? null : getReportStatus(status!);

  factory MovieIssueReport.fromJson(Map<String, dynamic> json) =>
      _$MovieIssueReportFromJson(json);

  Map<String, dynamic> toJson() => _$MovieIssueReportToJson(this);
}
